### PR TITLE
Planning: enhance several corner case in path_bounds_decider

### PR DESCRIPTION
**Pull request by xmyqsh**

### CODE REVIEW

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc [hunk 1]
```diff
@@ -392,15 +392,16 @@ Status PathBoundsDecider::GeneratePullOverPathBound(
...
+         curr_idx + kNumExtraTailBoundPoint) {
     path_bound->pop_back();
   }
-  for (int idx = 0; idx < kNumExtraTailBoundPoint; ++idx) {
-    std::get<1>((*path_bound)[path_bound->size() - 1 - idx]) =
-        std::get<1>((*path_bound)[std::get<3>(pull_over_configuration)]);
-    std::get<2>((*path_bound)[path_bound->size() - 1 - idx]) =
-        std::get<2>((*path_bound)[std::get<3>(pull_over_configuration)]);
+  for (int idx = curr_idx + 1;
+       idx < std::min(static_cast<int>(path_bound->size()),
```
Comment (jmtao) [inline modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: comment same as above

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc [hunk 2]
```diff
@@ -392,15 +392,16 @@ Status PathBoundsDecider::GeneratePullOverPathBound(
...
+                      curr_idx + kNumExtraTailBoundPoint + 1); ++idx) {
+    std::get<1>((*path_bound)[idx]) = std::get<1>((*path_bound)[curr_idx]);
+    std::get<2>((*path_bound)[idx]) = std::get<2>((*path_bound)[curr_idx]);
   }
```
Comment (jmtao) [inline modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: you are not really changing anything, right?

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc [hunk 3]
```diff
@@ -559,9 +560,10 @@ bool PathBoundsDecider::SearchPullOverPosition(
           (0.5 * (kPulloverLonSearchCoeff - 1.0) * vehicle_param.length() +
            vehicle_param.back_edge_to_center()) /
           vehicle_param.length() / kPulloverLonSearchCoeff;
-      const auto& pull_over_point = path_bound[static_cast<size_t>(
-          back_clear_to_total_length_ratio * static_cast<double>(i) +
-          (1.0 - back_clear_to_total_length_ratio) * static_cast<double>(j))];
+      auto idx = static_cast<size_t>(
```
Comment (panjiacheng) [inline modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: Please try to give it a more meaningful name, such as "pull_over_idx", etc.

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc [hunk 4]
```diff
@@ -351,11 +351,11 @@ Status PathBoundsDecider::GeneratePullOverPathBound(
              curr_idx + kNumExtraTailBoundPoint) {
         path_bound->pop_back();
       }
-      for (int idx = 0; idx < kNumExtraTailBoundPoint; ++idx) {
-        std::get<1>((*path_bound)[path_bound->size() - 1 - idx]) =
-            std::get<1>((*path_bound)[curr_idx]);
-        std::get<2>((*path_bound)[path_bound->size() - 1 - idx]) =
-            std::get<2>((*path_bound)[curr_idx]);
+      for (int idx = curr_idx + 1;
+           idx < std::min(static_cast<int>(path_bound->size()),
```
[the remark this answers is not in the document]
Comment (xmyqsh) [reply modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: Done.

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc (hunk 1 above)
Comment (xmyqsh) [reply modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: Done.

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc (hunk 2 above)
Comment (xmyqsh) [reply modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: @jmtao 
Here, I have fixed up the corner case when `path_bound->size() < kNumExtraTailBoundPoint` for runtime error and when `curr_idx + kNumExtraTailBoundPoint > path_bound->size()` for logic error.

File: modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc (hunk 3 above)
Comment (xmyqsh) [reply modules/planning/tasks/deciders/path_bounds_decider/path_bounds_decider.cc]: Done.

**Comment by xmyqsh** [comment]
@jmtao @panjiacheng 
Two main things have been fixed in this PR.
First, I have fixed up the corner case when path_bound->size() < kNumExtraTailBoundPoint for runtime error and when curr_idx + kNumExtraTailBoundPoint > path_bound->size() for logic error.

Second, I have fixed and unified the `pull_over_idx ` calculation from `(i + j) / 2` to `static_cast<size_t>(
          back_clear_to_total_length_ratio * static_cast<double>(i) +
          (1.0 - back_clear_to_total_length_ratio) * static_cast<double>(j))`.

**Pull request merged**